Binding documentation

kindofluke opened the pull request:
This pull request is to facilitate discussion of the binding documentation for which documentation help was requested in #1060 


this has been my approach: 

- [x] With adding input elements and bindings the page should just be called `interactions`. 
- [x] Add simple dropdown example
- [x] add an example using filter
- [x] enumerate all of the avail input elements. 
- [x] bind to  "scales" magic for panning and zooming
- [x] headers are all out of wack
- [x] spelling and proofread 

I am attaching a small screen recording of the two examples: 


![binding_small](https://user-images.githubusercontent.com/20745904/44551801-3370db80-a6f6-11e8-8183-f9d451a25490.gif)


Opening the PR for general feedback I don't think its ready but I need some ideas on next steps.

kindofluke commented:
@jakevdp I also was going to add this as just a good interaction example. (you are the original author of the [vega example](https://vega.github.io/vega-lite/examples/interactive_seattle_weather.html) but I can't quite get the interval select to work when you select from the scatter and the bars adjust.


[GIST Seattle Weather](https://gist.github.com/kindofluke/e38d7e733f7aa4f3cf112d5274ff0570)

kindofluke commented:
I totally agree. I will try to make a compound chart that has drop down radio buttons, and the rest.

jakevdp commented:
We do have a version of that in the gallery: https://altair-viz.github.io/gallery/seattle_weather_interactive.html

kindofluke commented:
🤦 totally missed this.

kindofluke commented:
Do you think it is possible to change the x or y fields through a selection? 

I was thinking of this example [Scatter Plot With Varying Dimension](https://vega.github.io/vega/examples/scatter-plot-null-values/
). 

I think this is possible with with an expression referring to the signal value. If it is possible it could be really helpful as example of a selection not referring to an encoding. If not, I am still going to use this movie dataset because it's rich scatter plot that benefits from filters and highlights based on inputs. 

I can do radio buttons to highlight movies of a certain imdb rating. Maybe a slider to filter the gross total of each film. I probably won't recreate the null values though that is a nice feature.

jakevdp commented:
> Do you think it is possible to change the x or y fields through a selection?

No, that's not possible in Vega-Lite.

jakevdp commented:
You can work around it by stacking the two columns, adding an indicator variable, and filtering on that indicator variable.

kindofluke commented:
Addressed most of your comments in the latest and added an example based on movies. 

the example is having a problem with a radio buttons (at least on my browser) I am going to look at that tomorrow (getting late)

<img width="516" alt="image" src="https://user-images.githubusercontent.com/20745904/44561364-7e084d00-a722-11e8-96af-272f22b399b9.png">

jakevdp commented:
One of the issues with bindings is that they are not initialized properly, and do not apply any selection or filtering until the user interacts with them. This is particularly problematic in the case of multiple bindings, because you have to click all of them at least once before they will interact properly. That's one reason I haven't done much with docs or examples of bindings up to this point.

It's being addressed in vega-lite... I can't find the issue at the moment.

jakevdp commented:
Yeah, we try to avoid anything requiring HTTP access in the example scripts, so that tests can be run in contexts with no web access.

Vega-Lite will parse the date properly if it's used in a temporal encoding within the chart.

If you're not using the column in an encoding but you still want it parsed as a date, you can use the ``alt.UrlData(url, format=alt.DataFormat(parse={'Release_Date': 'date'))``

kindofluke commented:
OK Everything is all set now I think. 

We have a solid example of all the bindings and a decent write up of how the bindings actually work and connect to the selections. 

Let me know what you think happy to keep tinkering or we can brainstorm some more examples.

kindofluke commented on doc/user_guide/interactions.rst:
```diff
@@ -236,9 +251,101 @@ over them with your mouse:
+
+Bindings and input elements can also be used to filter data on the client side. Reducing noise in the chart and allowing the user to see just certain selected elements:
+
+.. altair-plot::
```
great catch. I only found one other reference (on the weather chart which was funny given that I missed it before) I think we should be there.

kindofluke commented on doc/user_guide/interactions.rst:
```diff
@@ -1,19 +1,31 @@
 .. currentmodule:: altair
 
-.. _user-guide-selections:
+.. _user-guide-interactions:
```
Sorry I missed this in previous. Updated found one reference as stated above.

kindofluke commented:
All set on latest commit. If I am too much trouble you can throw me another one from #1060.

jakevdp commented:
Looks good – thanks so much for all your work on this!
